src/server/models/Player.js

```diff
@@ -324,13 +339,17 @@ export default class Player extends Model {
     while (this.nativeVillageMat.size > 0) {
       this.moveCard(this.hand.last(), this.nativeVillageMat, this.deck);
     }
+    this.deck.forEach(c => {
+      c.endGameCleanUp(this);
+    });
   }
 
   async play(card) {
     this.game.log(`${this.name} plays ${card.name}`);
     this.cardsPlayedThisTurn.push(card);
     this.moveCard(card, this.hand, this.playArea);
     await this.emit('play', card, this);
+    if (card.types.has('Duration')) card.ignoreCleanUp = true;
```

Comment:
this might be better inside the `onPlay` of every card

Reply:
I didn't want to have to do it to all durations since it leads to error, it's much harder to forget to set the flag if it's automatically set here.

Reply:
in Software Engineering terms this is a conflict between [DRYness](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself) and [Separation of Concerns](https://en.wikipedia.org/wiki/Separation_of_concerns).  In this case I'd say that Separation of Concerns is more important -- the less `Player.js` knows about how cards work, the better.  If every new card type had a special line here, then this would get messy quick